### PR TITLE
[Perf] Downgrade mxfp4 triton_kernels to 3.5

### PR DESCRIPTION
**Pull request by xyang16**
## Purpose

triton_kernels was upgraded to 3.6.0 in #30525. This caused more than 30% performance regression at batch size 1, because triton_kernels 3.6.0 has much worse performance than 3.5. So this PR downgrade triton_kernels back to 3.5.1.

## Test Plan

```
pytest -s -v tests/kernels/moe/test_gpt_oss_triton_kernels.py
```

## Benchmarking

Benchmarked on H200.

```
vllm serve /opt/dlami/nvme/models/gpt-oss-20b \
  --tensor-parallel-size 1 \
  --max-num-seqs 16 \
  --no-enable-prefix-caching
```

Main:

concurrency=1

```
============ Serving Benchmark Result ============
Successful requests:                     60        
Failed requests:                         0         
Maximum request concurrency:             1         
Benchmark duration (s):                  75.04     
Total input tokens:                      15599     
Total generated tokens:                  18000     
Request throughput (req/s):              0.80      
Output token throughput (tok/s):         239.87    
Peak output token throughput (tok/s):    243.00    
Peak concurrent requests:                2.00      
Total token throughput (tok/s):          447.75    
---------------Time to First Token----------------
Mean TTFT (ms):                          17.94     
Median TTFT (ms):                        17.85     
P99 TTFT (ms):                           28.54     
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          4.12      
Median TPOT (ms):                        4.12      
P99 TPOT (ms):                           4.13      
---------------Inter-token Latency----------------
Mean ITL (ms):                           4.12      
Median ITL (ms):                         4.12      
P99 ITL (ms):                            4.44      
==================================================
```

concurrency=16

```
============ Serving Benchmark Result ============
Successful requests:                     960       
Failed requests:                         0         
Maximum request concurrency:             16        
Benchmark duration (s):                  137.76    
Total input tokens:                      217301    
Total generated tokens:                  288000    
Request throughput (req/s):              6.97      
Output token throughput (tok/s):         2090.54   
Peak output token throughput (tok/s):    2448.00   
Peak concurrent requests:                32.00     
Total token throughput (tok/s):          3667.89   
---------------Time to First Token----------------
Mean TTFT (ms):                          123.38    
Median TTFT (ms):                        74.02     
P99 TTFT (ms):                           742.44    
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          7.26      
Median TPOT (ms):                        7.09      
P99 TPOT (ms):                           9.45      
---------------Inter-token Latency----------------
Mean ITL (ms):                           7.26      
Median ITL (ms):                         6.97      
P99 ITL (ms):                            9.70      
==================================================
```

PR:

concurrency=1

```
============ Serving Benchmark Result ============
Successful requests:                     60        
Failed requests:                         0         
Maximum request concurrency:             1         
Benchmark duration (s):                  55.57     
Total input tokens:                      15599     
Total generated tokens:                  18000     
Request throughput (req/s):              1.08      
Output token throughput (tok/s):         323.91    
Peak output token throughput (tok/s):    327.00    
Peak concurrent requests:                3.00      
Total token throughput (tok/s):          604.61    
---------------Time to First Token----------------
Mean TTFT (ms):                          17.90     
Median TTFT (ms):                        17.82     
P99 TTFT (ms):                           28.78     
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          3.04      
Median TPOT (ms):                        3.04      
P99 TPOT (ms):                           3.05      
---------------Inter-token Latency----------------
Mean ITL (ms):                           3.04      
Median ITL (ms):                         3.04      
P99 ITL (ms):                            3.41      
==================================================
```

concurrency=16

```
============ Serving Benchmark Result ============
Successful requests:                     960       
Failed requests:                         0         
Maximum request concurrency:             16        
Benchmark duration (s):                  118.07    
Total input tokens:                      217301    
Total generated tokens:                  288000    
Request throughput (req/s):              8.13      
Output token throughput (tok/s):         2439.18   
Peak output token throughput (tok/s):    2944.00   
Peak concurrent requests:                32.00     
Total token throughput (tok/s):          4279.59   
---------------Time to First Token----------------
Mean TTFT (ms):                          132.54    
Median TTFT (ms):                        39.41     
P99 TTFT (ms):                           653.21    
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          6.14      
Median TPOT (ms):                        5.92      
P99 TPOT (ms):                           7.95      
---------------Inter-token Latency----------------
Mean ITL (ms):                           6.14      
Median ITL (ms):                         5.78      
P99 ITL (ms):                            10.37     
==================================================
```

## Accuracy Testing

```
OPENAI_API_KEY=EMPTY python3 -m gpt_oss.evals --model /opt/dlami/nvme/models/gpt-oss-20b --eval gpqa --n-threads 200 --reasoning-effort low
```

Main:

```
{'chars': np.float64(60.72348484848485), 'chars:std': np.float64(222.8212538462918), 'score': np.float64(0.5669191919191919), 'score:std': np.float64(0.4955015860245882)}
Writing results to /tmp/gpqa___opt__dlami__nvme__models__gpt-oss-20b-low_temp1.0_20260406_235002.json
Writing all results to /tmp/gpqa___opt__dlami__nvme__models__gpt-oss-20b-low_temp1.0_20260406_235002_allresults.json
[{'eval_name': 'gpqa', 'model_name': '__opt__dlami__nvme__models__gpt-oss-20b-low_temp1.0_20260406_235002', 'metric': 0.5669191919191919}]
```

PR:

```
{'chars': np.float64(66.51010101010101), 'chars:std': np.float64(239.0948997994119), 'score': np.float64(0.571969696969697), 'score:std': np.float64(0.49479325249854617)}
Writing results to /tmp/gpqa___opt__dlami__nvme__models__gpt-oss-20b-low_temp1.0_20260407_202524.json
Writing all results to /tmp/gpqa___opt__dlami__nvme__models__gpt-oss-20b-low_temp1.0_20260407_202524_allresults.json
[{'eval_name': 'gpqa', 'model_name': '__opt__dlami__nvme__models__gpt-oss-20b-low_temp1.0_20260407_202524', 'metric': 0.571969696969697}]
```
---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/layers/fused_moe/experts/gpt_oss_triton_kernels_moe.py [hunk 1]
```diff
@@ -159,22 +136,17 @@ def triton_kernel_moe_forward(
...
+
+        logits = gating_output
+        if sm_first:
+            logits = torch.softmax(logits, dim=-1)
+        topk_weights, topk_ids_raw, _ = topk_fn(
+            logits, topk, apply_softmax=not sm_first
+        )
+
         # topk_ids_raw contains global expert IDs - remap to local.
         topk_ids = expert_map[topk_ids_raw.to(torch.long)]
```
[the remark this answers is not in the document]
Comment (xyang16) [reply vllm/model_executor/layers/fused_moe/experts/gpt_oss_triton_kernels_moe.py:150]: It's the same as the old code. Not sure whether topk_ids_raw will contain -1.
